ci: clang-tidy fails and no obvious way to fix

CI seems to be having trouble with code/BUILD files that look correct.

One example is with failure to find a protobuf-generated include file, visible in https://github.com/envoyproxy/envoy/pull/10569 . The same PR also fails clang-tidy with a suggestion to use `= default` for a trivial constructor, but the suggestion does not appear to be correct as the constructor requires an initializer.
```
test/integration/filter_manager_integration_test.cc:14:10: error: 'test/integration/filter_manager_integration_test.pb.h' file not found [clang-diagnostic-error]
#include "test/integration/filter_manager_integration_test.pb.h"
         ^
/build/tmp/_bazel_bazel/b570b5ccd0454dc9af9f65ab1833764d/execroot/envoy/test/integration/filter_manager_integration_test.cc:444:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
  InjectDataWithEchoFilterIntegrationTest()
  ^
/build/tmp/_bazel_bazel/b570b5ccd0454dc9af9f65ab1833764d/execroot/envoy/test/integration/filter_manager_integration_test.cc:489:3: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
  InjectDataWithTcpProxyFilterIntegrationTest()
  ^
```

I have seen other instances too, unrelated to protobufs, where an include file cannot be found, though it's present in the other CI steps and for local builds. @antoniovicente found the same issue with `benchmark.h` in https://github.com/envoyproxy/envoy/pull/10538

--- a/tools/gen_compilation_database.py
+++ b/tools/gen_compilation_database.py
@@ -10,9 +10,15 @@ from pathlib import Path
 
 
 def runBazelBuildForCompilationDatabase(bazel_options, bazel_targets):
-  query = 'attr(include_prefix, ".+", kind(cc_library, deps({})))'.format(
-      ' union '.join(bazel_targets))
-  build_targets = subprocess.check_output(["bazel", "query", query]).decode().splitlines()
+  query_targets = ' union '.join(bazel_targets)
+  query = ' union '.join(
+      q.format(query_targets) for q in [
+          'attr(include_prefix, ".+", kind(cc_library, deps({})))',
+          'attr(strip_include_prefix, ".+", kind(cc_library, deps({})))',
+          'attr(generator_function, ".*proto_library", kind(cc_.*, deps({})))',
+      ])
+  build_targets = subprocess.check_output(["bazel", "query", "--notool_deps",
+                                           query]).decode().splitlines()
   subprocess.check_call(["bazel", "build"] + bazel_options + build_targets)
 
 

